python/grass/jupyter/interact_display.py

```diff
@@ -142,11 +144,71 @@ def add_vector(self, name):
             input=name,
             output=json_file,
             format="GeoJSON",
-            env=self._vector_env,
+            env=self._wgs84_env,
         )
         # Import GeoJSON to folium and add to map
         folium.GeoJson(str(json_file), name=name).add_to(self.map)
 
+    def add_raster(self, name, opacity=0.8):
+        """Imports raster into temporary WGS84 location,
+        exports as png and overlays on folium map
+
+        :param str name: name of raster to add to display
+        :param float opacity: raster opacity, number between
+                              0 (transparent) and 1 (opaque)
+        """
+
+        # Find full name of raster
+        file_info = gs.find_file(name, element="raster")
+        full_name = file_info["fullname"]
+        name = file_info["name"]
+
+        # Reproject raster into WGS84/epsg3857 location
+        env_info = gs.gisenv(env=self._src_env)
+        resolution = estimate_resolution(
+            full_name, env_info["GISDBASE"], env_info["LOCATION_NAME"], self._psmerc_env
+        )
+        gs.run_command(
+            "r.proj",
+            input=full_name,
+            output=name,
+            location=env_info["LOCATION_NAME"],
+            dbase=env_info["GISDBASE"],
+            resolution=resolution,
+            env=self._psmerc_env,
+        )
+
+        # Reprojects bounds of raster for overlaying png (THIS HAS TO BE IN WGS84)
+        bounds = gs.read_command("g.region", flags="g", env=self._src_env)
+        bounds = gs.parse_key_val(bounds, sep="=", vsep="\n")
+        proj = gs.read_command("g.proj", flags="jf", env=self._src_env)
+        north, east = convert_coordinates_to_latlon(bounds["e"], bounds["n"], proj)
+        south, west = convert_coordinates_to_latlon(bounds["w"], bounds["s"], proj)
+        new_bounds = [[north, west], [south, east]]
+
+        # Write raster to png file with GrassRenderer
+        filename = os.path.join(self.tmp_dir, "map.png")
```

Comment:
I'm still not sure I fully understand why passing the original name could cause a problem here. Are you meaning that `elevation@PERMANENT` and `elevation@user1` are different rasters but we want to add both of them to our folium map. In that case, we'd end up with two `elevation`s in the temporary location which would cause a problem. Perhaps replacing the `@` in the `full_name` string to make something like `elevation_user1` as the new name would solve this?

Reply:
I see two issues:

1. `self.tmp_dir` + `"map.png"` will be the same filename for each `add_raster` call. Is this a problem? For GrassRenderer? For folium.ImageOverlay`? I have no idea how ImageOverlay behaves or what is in its doc. When this image needs to be present? Using the new name (from 2.) might be the best option at this point.
2. Replacing `@` with `_` should work well enough. We will always do that so that should avoid any conflicts (mapname@mapsetname is a unique identifier for a map of a given type within a location). (If we would convert rasters to vectors or the other way around - like in the possible alternative workflows we talked about - that would create potential for conflicts, but we could avoid that by simply using different location or different mapset for that workflow.)